### PR TITLE
Fix unknwon logFile issue + add documentation on logDir option

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

The `logDir` option wasn't documented properly. Also there was a bug in #3173 where the main reporter tried to create a log file with unknown log file name. This patch prevents this from happening and adds more documentation.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-junit-reporter/README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ module.exports = {
     reporters: [
         'dot',
         ['junit', {
-            outputDir: './',
+            logDir: './',
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply packages/wdio-junit-reporter/README.md]: What name for this property would you suggest then?

[the remark this answers is not in the document]
Comment (BorisOsipov) [reply packages/wdio-junit-reporter/README.md]: `outputDir` - at least it looks like in v4 and will make migration seamless

[the remark this answers is not in the document]
Comment (christian-bromann) [reply packages/wdio-junit-reporter/README.md]: agree, renamed config back to `outputDir`

File: examples/wdio/mocha/wdio.conf.js [hunk 2]
```diff
@@ -25,9 +25,9 @@ exports.config = {
     logLevel: 'trace',
     coloredLogs: true,
     framework: 'mocha',
-    logDir: __dirname,
+    // logDir: __dirname,
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply examples/wdio/mocha/wdio.conf.js]: was for testing purposes, reverted change

**Comment by christian-bromann** [comment]
@BorisOsipov addressed issues, please review again

**Pull request merged**